clippy_lints/src/misc.rs

```diff
@@ -519,7 +519,13 @@ fn is_signum(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
 }
 
 fn is_float(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
-    matches!(walk_ptrs_ty(cx.tables.expr_ty(expr)).sty, ty::Float(_))
+    let value = &walk_ptrs_ty(cx.tables.expr_ty(expr)).sty;
+
+    if let ty::Array(arr_ty, _) = value {
+        return matches!(arr_ty.sty, ty::Float(_));
```

Comment:
I think we also have to `walk_ptrs_ty` on `arr_ty`, so that Arrays of `&float`s also get linted.

Reply:
On it.

Reply:
I'm having a bit of an issue getting the `walk_ptrs_ty` to work with `arr_ty`; I have updated the tests in the meanwhile. You can have a look if I'm going in the right direction so far.